tests/DoctrineAuditBundle/Helper/DoctrineHelperTest.php

```diff
@@ -11,18 +12,31 @@
  */
 final class DoctrineHelperTest extends TestCase
 {
-    public function testGenerateProxyClassName(): void
-    {
-        self::assertSame('DH\DoctrineAuditBundle\Tests\Fixtures\Core\Standard\__CG__\Post', DoctrineHelper::generateProxyClassName('Post', mb_substr(Post::class, 0, mb_strrpos(Post::class, '\\'))));
-    }
-
-    /**
-     * @depends testGenerateProxyClassName
-     */
     public function testGetRealName(): void
     {
-        self::assertNotSame('Post', DoctrineHelper::getRealClass(Post::class));
-        self::assertSame(Post::class, DoctrineHelper::getRealClass(Post::class));
-        self::assertSame(Post::class, DoctrineHelper::getRealClass(DoctrineHelper::generateProxyClassName(Post::class, mb_substr(Post::class, 0, mb_strrpos(Post::class, '\\')))));
+        self::assertNotSame('Post', DoctrineHelper::getRealClassName(Post::class));
+        self::assertNotSame('Comment', DoctrineHelper::getRealClassName(Comment::class));
+
+        self::assertSame(Post::class, DoctrineHelper::getRealClassName(Post::class));
+        self::assertSame(Comment::class, DoctrineHelper::getRealClassName(Comment::class));
+
+        self::assertSame('DH\DoctrineAuditBundle\Tests\Fixtures\Core\Standard\Post', DoctrineHelper::getRealClassName(Post::class));
+        self::assertSame('DH\DoctrineAuditBundle\Tests\Fixtures\Core\Standard\Comment', DoctrineHelper::getRealClassName(Comment::class));
+
+        $post = new Post();
+        self::assertSame(Post::class, DoctrineHelper::getRealClassName($post));
+
+        $comment = new Comment();
+        self::assertSame(Comment::class, DoctrineHelper::getRealClassName($comment));
+
+        // __CG__: Doctrine Common Marker for Proxy (ODM < 2.0 and ORM < 3.0)
+        self::assertSame(Post::class, DoctrineHelper::getRealClassName('Proxies\__CG__\DH\DoctrineAuditBundle\Tests\Fixtures\Core\Standard\Post'));
```

Comment:
In fact the `DoctrineHelper::generateProxyClassName()` method was only there to help not to write such FQCN.
BTW, as it wasn't used elsewhere, it's ok to have it removed.